specs/WebResourceResponseReceived.md

```diff
@@ -0,0 +1,219 @@
+            CHECK_FAILURE(args->get_Request(&webResourceRequest));
+            // The response object as received
+            wil::com_ptr<ICoreWebView2WebResourceResponse> webResourceResponse;
+            CHECK_FAILURE(args->get_Response(&webResourceResponse));
+            
+            // Get body content for the response. Redirect responses will
+            // return an error HRESULT as their body (if any) is ignored.
+            HRESULT populateCallResult = args->PopulateResponseContent(
+                Callback<
+                    ICoreWebView2WebResourceResponseReceivedEventArgsPopulateResponseContentCompletedHandler>(
+                    [this, webResourceRequest, webResourceResponse](HRESULT result) {
+                        // The response might not have a body.
+                        bool populatedBody = SUCCEEDED(result);
+                        
+                        std::wstring message =
+                            L"{ \"kind\": \"event\", \"name\": "
+                            L"\"WebResourceResponseReceived\", \"args\": {"
+                            L"\"request\": " +
+                            RequestToJsonString(webResourceRequest.get()) +
+                            L", "
+                            L"\"response\": " +
+                            ResponseToJsonString(webResourceResponse.get()) + L"}";
+
+                        message +=
+                            WebViewPropertiesToJsonString(m_webview.get());
+                        message += L"}";
+                        PostEventMessage(message);
+                        return S_OK;
+                    })
+                    .Get());
+
+            return S_OK;
+        })
+        .Get(),
+    &m_webResourceResponseReceivedToken);
+```
+
+## C#
+```c#
+WebView.WebResourceResponseReceived += WebView_WebResourceResponseReceived;
+
+// Note: modifications made to request and response are ignored
+private async void WebView_WebResourceResponseReceived(object sender, CoreWebView2WebResourceResponseReceivedEventArgs e)
+{
+    // Actual headers sent with request
+    foreach (var current in e.Request.Headers)
+        Console.WriteLine(current);
+
+    // Headers in response received
+    foreach (var current in e.Response.Headers)
+        Console.WriteLine(current);
+
+    // Status code from response received
+    int status = e.Response.StatusCode;
+    if (status == 200)
+    {
+        // Handle
+        Console.WriteLine("Request succeeded!");
+
+        // Get response body
+        try
+        {
+            await e.PopulateResponseContentAsync();
+            DoSomethingWithResponseBody(e.Response.Content);
+        } catch (Exception ex)
+        {
+            // An exception will be thrown if the request has no body.
+        }
+    }
+}
+```
+
+
+# Remarks
+Calling `PopulateResponseContent` will fail/throw an exception if the response
+has no body. An exception will also be thrown for redirect responses, for which
```

Comment:
Do you mean if the response is an HTTP redirect? From [HTTP spec](https://tools.ietf.org/html/rfc2616#section-4.3) it seems like only 1xx, 204, and 304 must not have a response body and other forms of redirect are allowed to have a response body?

Reply:
While other forms of redirect are allowed to have a response body, the network stack will ignore that body so they will be treated as having no body. The exception thrown is the same as for a response without a body. I reworded to better reflect that.